### PR TITLE
📚 Docs: Guide - Hello World & Generative Shape Art

### PR DESCRIPTION
**Pull request by dannyfritz**
📚📚📚

**Benefits of a Guide:**
- More approachable and friendly than code samples
- See the process of coding something. Not just the finished product.
- Common pitfalls explained and avoided
- Explanations of trade offs
- Conceptual peeks under the hood of `ggez`'s modules for deeper understanding

**Assumptions and Guidelines:**
- User can write a hello world in Rust
- User hasn't done anything with `ggez`
- User has very little gamedev experience
- Written from the `we`/`us` perspective unless an action must betaken, then `you`/`your` is used
- Formatted with Markdown
- Written in English
- Conceptually, earlier sections should be easier and later sections should be harder
- Tone is fun, friendly, and joking
- Always link to the `ggez` docs when appropriate
- Include unguided challenges at the end of each guide
- Have a brief summary at the top explaining what the User can do after reading the guide

**Guide Sections:**
- [x] Hello `ggez`! (Loop + Window + Conf)
  - Prerequisites
  - Loop
  - Window / Conf
- [x] Generative Art (Shapes)

**For another PR**
- [ ] Overview
  - What is `ggez`?
  - How is it different than X?
- [ ] Mouse Tail (Mouse)
- [ ] Menu (Text)
- [ ] Guessing Game (Keyboard)
- [ ] Image Viewer (Image)
- [ ] Music Player (Audio)
- [ ] Flappy Bird (Physics + AABB + Math)
- [ ] Mini Games (States)
- [ ] Image Filters (Shaders)

📚📚📚

### CODE REVIEW

File: docs/guides/HelloWorld.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+[dependencies]
+ggez = "0.4"
+```
+
+### Check Project Setup ✔
+
+Test to make sure everything is correct by running `cargo run`.
+
+Be patient. This can take a while on the first run. Subsequent builds will be much faster.
+Be sure it says `Hello, world!` at the bottom. before continuing.
```
Comment (0nkery) [inline docs/guides/HelloWorld.md]:
```diff
- at the bottom. before continuing.
+ at the bottom, before continuing.
```
I'm not much into English grammar, but it feels like there's no need in comma at all.

Comment (dannyfritz) [reply docs/guides/HelloWorld.md]: Probably doesn't need any punctuation at all there. 😄

File: docs/guides/HelloWorld.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+### Loop
+
+All real time games have a loop.
+Often this loop is referred to as a game loop.
+This loop is responsible for:
+0. Handling events
+0. Updating state
+0. Drawing
+
+`ggez` provides [`Trait ggez::Event::EventHandler`](https://docs.rs/ggez/0.4.0/ggez/event/trait.EventHandler.html) is the provided loop as a loop for us to use in our games.
```
Comment (0nkery) [inline docs/guides/HelloWorld.md]: `ggez provides EventHandler as a loop ...`
or
`EventHandler is the provided loop for us to use` (but I'd prefer former).

**Comment by 0nkery** [comment]
Good start! I believe you'll make an awesome thing.

**Comment by icefoxen** [comment]
I like your roadmap.  Things we need to think about from a project perspective (mostly just recording what we talked about on IRC):

 * We need to ensure the guide is maintained and updated as ggez continues evolving and new features get added -- this is NOT slightly
 * It would be nice to be able to ensure all code in it is correct, via [rust-skeptic](https://github.com/budziq/rust-skeptic) or some other means
 * It would be nice to be able to put the guide on the website, and have the website be the actual entry point to useful information rather than the github page
 * We should at least think about the possibility of translations and structure it so they're easy to do if we ever get around to it.
 * Do we want to make this **the** ggez guide, and commit to keeping it up to date, or settle for it having, I guess, "contrib" status where keeping it up to date isn't a show-stopper for making a release?  I definitely want high-quality official docs, but also, getting everything shipshape for an official release is *hard*, and is only going to become more work as time goes on...

File: docs/guides/HelloGgez.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+All games have a loop.
+Often the loop is referred to as a game loop.
+
+Game loops are responsible for:
+
+1. Handling events such as keyboard, mouse, closing window, etc.
+1. Updating state such as player position, health, etc.
+1. Drawing shapes, images, etc.
+
+[`EventHandler`](https://docs.rs/ggez/0.4.0/ggez/event/trait.EventHandler.html) is a loop provided by `ggez` to use in our games.
```
Comment (termhn) [inline docs/guides/HelloGgez.md]: I think this is better put as, "EventHandler is a way for us to interface with--and implement our own logic on top of--ggez's internal game loop." or something similar.

**Comment by termhn** [comment]
Sweet! I like the direction this is going and am definitely interested in helping out as it progresses! :D

**Comment by itmuckel** [comment]
I like your style of writing (and I love emoticons). 

The only thing I'm missing is integration with specs-rs, but as soon as I'm proficient enough in this I consider contributing a part. (I'm still developing a game that makes excessive use of specs-rs). Also this is not too important.

Regarding keeping it up to date: I think it's generally best-practice to have a PR contain changes to the docs as soon as it adds functionality or (more important) changes functionality. I think this is easier to follow, because changes to code and changes to docs are in the same commit/PR. 😄

**Comment by icefoxen** [comment]
@dannyfritz What's the status of this?  Can we merge what's here so that other people can work on it?

**Comment by dannyfritz** [comment]
Yeah, you can merge this. I have a third one in the hopper I just need to format I can do another PR for later. Been going really slow as you can see. 😛

**Comment by icefoxen** [comment]
Awesome, thank you!  No worries.  Perhaps we can do one PR per chapter in the future, to keep things easier to put together piece by piece.

**Pull request merged**